doc/spec_TransferElement.dox

```diff
@@ -240,29 +240,30 @@ This documnent is currently still **INCOMPLETE**!
 
+- \anchor transferElement_comment_B_8_6_6 \ref transferElement_B_8_6_6 "8.6.6" 
+The current default implementation in the TransferElement base class is always throwing a ChimeraTK::logic_error to inform the programmer that the function has to be overridden for all implementations with wait_for_new_data.
+
 - \anchor transferElement_comment_B_9a \ref transferElement_B_9 "9." The ChimeraTK::runtime_error is the only exception that conceptually is recoverable. This is done by calling DeviceBackend::open(). Other exceptions are not recoverable by the running application:
   - ChimeraTK::logic_error is a programming or configuration mistake.
   - The boost::numeric::bad_numeric_cast conceptually also is a logical error (wrong data type) which only shows at run time when the data is overflowing.
   - The boost::thread_interrupted is thrown when TransferElement::interrupt() is called. It means that a blocking read() has been interrupted on request. It is not an error and hence there is nothing to recover.
 - \anchor transferElement_comment_B_9b \ref transferElement_B_9 "9." It does not matter if the exception occured in an asynchronous or synchronous read, or in a write operation.
-- \anchor transferElement_comment_B_9_1 \ref transferElement_B_9_1 "9.1" It depends on the implementation whether the backend already has done 9.2 and 9.3 when the transfer elements
+- \anchor transferElement_comment_B_9_1 \ref transferElement_B_9_1 "9.1" It depends on the implementation whether the backend already has done 9.3 and 9.4 when the transfer elements
   first sees the exeption and then reports it back again via DeviceBackend::setException(), or if it only happend in that function. The important part is
-  that meta-backends and the user application can trigger this situation (see \ref transferElement_B_10_1_1 "10.1.1" and \ref transferElement_B_10_1_3 "10.1.3")
-- 9.2.1 Open can be called again on an already opened backend to start error recovery.
-- 9.2.2 If an asynchronous read transfer is the first one to detect the exception, the implementation must make sure that it is only pushed once into the queue, and informing "all" transfer elememnts with wait_for_new data does not send it again if it was already put into the queue.
-- 9.2.4 Avoid race conditions here. The call to Device::activateAsyncRead() usually is done from a different thread than the transfer which caused the exception.
+  that meta-backends and the user application can trigger this situation (see \ref transferElement_B_9_2_1 "9.2.1" and \ref transferElement_B_9_2_3 "9.2.3")
+- \anchor transferElement_comment_B_9_1_4 \ref transferElement_B_9_1_4 "9.1.4" This functionality is already implemented in the NDRegisterAccessorDecorator base class. It does not apply for decorator-like TransferElements which are created by backends, because the backend has to call TransferElement::setExceptionBackend after the creation anyway (see \ref transferElement_B_9_1_1 "9.1.1").
```

Comment:
Ah, *all* Backends have to call setExceptionBackend on all accessors they create. The exception backend is not known in the constructor.
And if you do this on a decorator-like backend this will automatically propagate to the target, which is also implemented in the decorator base class.

Would it be better to have the exception backend as a constructor argument of the TransferElement, so you can set it in the constructor?

Reply:
No that's not true. Usually backends pass a shared pointer to themselves to the accessor upon creation. The accessor will assume that this is the exception backend. Only meta-backends need to call setExceptionBackend to override the exception backend of accessors produced by another backend.

Adding the exception backend as a constructor argument would simplify this approach.

Reply:
Right, each implementation can do that as is convenient.
So I will just take out the second, unnecessary and confusing sentence.

Reply:
> So I will just take out the second, unnecessary and confusing sentence.

No, this is also wrong. the thing you mentioned about setting the exception backend to the meta backend ist correct.